src/core/model.jl

```diff
@@ -37,21 +38,15 @@ space.
 """
 abstract type AbstractAgent end
 
-function correct_pos_type(n, model)
-    if typeof(model.space) <: GraphSpace
-        return coord2vertex(n, model)
-    elseif typeof(model.space) <: GridSpace
-        return vertex2coord(n, model)
-    end
-end
-
+abstract type AbstractSpace end
 SpaceType=Union{Nothing, AbstractSpace}
 
 struct AgentBasedModel{A<:AbstractAgent, S<:SpaceType, F, P}
     agents::Dict{Int,A}
     space::S
     scheduler::F
     properties::P
+    maxid::Vector{Int}
```

Comment:
Why is `maxid` not an `Int`?

Reply:
Yeah this is something to consider. Because the model type is not mutable and therefore you can't change it. Once this PR is done, and once we have benchmarking capabilities, I'll test a version where the model is mutable and `maxid` is Integer.

Reply:
Why not `maxid::Ref{Int}`?

Then call or set with `model.maxid[]`.